src/components/Navigation.astro

```diff
@@ -9,14 +9,30 @@ const pageData = {
     policies: "Policies",
     "about-us": "About"
 }
+const pathname = Astro.url.pathname;
+const basePath = pathname.split('/')[2];
```

Comment:
This doesn't look very future-proof. Is there a different way to do this?

Reply:
I think it depends on how we want to manage this. If we want to make pages handle their own navigational highlighting we would have to wire it through a bunch of pages, which might be messy. If we want to handle it here, we're essentially going to have to do _something_ that is doing a url comparison - which if we ever want to make changes to the base urls will require updates here - but we would need to do that anyway since the navigation is controlled by the pageData list above.

In short, i think this is basically as future proof as having the map in this file listing the navigation options. The front part of the path is at least consistent when generating astro pages on different environments - i had to deal with optional trailing / for the other navigation options that was massive pain.

Reply:
Fair enough. I don't think this is massively different than individual pages having local path links to other components or pages.